### ISSUE TITLE
Content-Encoding: deflate is handled incorrectly

### ISSUE THREAD

**Issue by chornyi** [spring-cloud/spring-cloud-netflix#394]
A header `Accept-Encoding: deflate, gzip` is appended to every origin request here:

https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java#L96

The origin server returns a `Content-Encoding: deflate`, but this header is removed from response here:

https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/ProxyRequestHelper.java#L158

There is  code to either decode `gzip` responses or add a `Content-Encoding: gzip` based on whether or not a client supports gzip:

https://github.com/spring-cloud/spring-cloud-netflix/blob/master/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendResponseFilter.java#L115-L129

However all other encodings are left as-is and no `Content-Encoding` header is sent to the client.

In summary:
* Any response that has `Content-Encoding` other than `gzip` will not be forwarded to the client correctly because that header is removed while encoding is preserved
* Spring Cloud specifically requests `Content-Encoding: deflate` which it then cannot handle correctly

This also makes me wonder: why does Spring need to get involved with content encodings at all? Could it preserve the response body and headers as-is and leave content negotiation to the client and origin server?

**Comment by spencergibb** [comment]
Closed via 80206e6b3861015438c0dec43ac8af100a10c1a2
